packages/models/src/actor/message-queue.ts

```diff
@@ -0,0 +1,76 @@
+/**
+ * @tutorial https://github.com/ckb-js/kuai/issues/17#issuecomment-1305896619
+ *
+ * The message queue is used as a set of actors' mailboxes.
+ * At this early stage, a custom queue is used for demostration, will be replaced by redis-stream
+ */
+/* eslint-disable-next-line @typescript-eslint/no-var-requires */
+const { scheduler } = require('node:timers/promises');
+
+import type { Registry } from './registry';
+import { Behavior } from '../utils';
+import { ActorURI, ActorMessage } from './interface';
+
+export class MessageQueue {
+  #mailboxes: Map<ActorURI, Array<ActorMessage>> = new Map();
+  #registry: Registry;
+  #isRunning = false;
+
+  constructor(registry: Registry) {
+    this.#registry = registry;
+  }
+
+  start = async (): Promise<void> => {
+    this.#isRunning = true;
+    while (this.#isRunning) {
+      await scheduler.wait(1000);
```

Comment:
Why 1000? Or should it be configurable?

Reply:
The `scheduler` in the custom message queue is for demonstrating asynchronous message consumption(jam in message queue) and it will be removed because messages should be handled instantly in fact.